presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -38,6 +39,19 @@ public PrestoSparkConfig setSparkPartitionCountAutoTuneEnabled(boolean sparkPart
         return this;
     }
 
+    @Config("spark.max-spark-input-partition-count-for-auto-tune")
+    @ConfigDescription("Max Spark input partition count when Spark partition auto tune is enabled")
```

Comment:
When auto tune is not enabled, do we also want to set a max limit ? actual partition count = min(count_set_by_session_property,  hard_limit_set_by_this_config)

Reply:
@viczhang861 : this is also used as session property. I feel we should allow user to manually specify some large number when they intend to do so. 

Although so many config properties is not ideal anyway, we should really implement history-based auto mapper tuning.